foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_11_0/DockerModulePomDependencyTypeMigration.java

```diff
@@ -0,0 +1,110 @@
+import static org.technologybrewery.baton.util.pom.LocationAwareMavenReader.START;
+
+/**
+ * This migration fixes the build cache dependency check by removing the dependency `pom` type for the java pipeline
+ * and replace the dependency type with `habushu` if it's python pipeline
+ */
+public class DockerModulePomDependencyTypeMigration extends AbstractPomMigration {
+    private static final String PIPELINE_GROUP_ID = "${project.groupId}";
+    private static final String JAVA_PIPELINE = "java";
+    private static final String PYTHON_PIPELINE = "python";
+    private static final String HABUSHU = "habushu";
+    private static final String POM = "pom";
+
+    @Override
+    protected boolean shouldExecuteOnFile(File pomFile) {
+        Model model = PomHelper.getLocationAnnotatedModel(pomFile);
+        if (model.getPackaging().equals("docker-build")) {
+            // find the pipeline artifacts
+            Map<String, String> pipelineArtifactIds = getPipelineArtifactIds(pomFile);
+            return getDependenciesNeedToBeUpdated(model, pipelineArtifactIds).size() > 0;
+        }
+        return false;
+    }
+
+    @Override
+    protected boolean performMigration(File pomFile) {
+        Model model = PomHelper.getLocationAnnotatedModel(pomFile);
+
+        PomModifications modifications;
+        Map<String, String> pipelineArtifactIds = getPipelineArtifactIds(pomFile);
+        List<Dependency> dependenciesToBeUpdated = getDependenciesNeedToBeUpdated(model, pipelineArtifactIds);
+
+        modifications = new PomModifications();
+        for (Dependency dependency: dependenciesToBeUpdated) {
+
+            if (POM.equals(dependency.getType())) {
+                if (pipelineArtifactIds.get(dependency.getArtifactId()).equals(JAVA_PIPELINE)) {
+                    // delete the dependency type if it's java pipeline
+                    modifications.add(deleteTag(dependency, "type"));
+                } else {
+                    // replace the dependency type if it's python pipeline
+                    modifications.add(replaceInTag(dependency, "type", HABUSHU));
+                }
+            } else if (pipelineArtifactIds.get(dependency.getArtifactId()).equals(PYTHON_PIPELINE)) {
+                int indentSize = dependency.getLocation("artifactId" + START).getColumnNumber() - 1;
+                String indent = StringUtils.repeat(' ', indentSize);
+                modifications.add(new PomModifications.Insertion(dependency.getLocation(END), 0,
+                        ignore -> indent + "<type>" + HABUSHU + "</type>\n"));
+            }
+        }
+        return PomHelper.writeModifications(pomFile, modifications.finalizeMods());
+    }
+
+    private static Map<String, String> getPipelineArtifactIds(File file) {
+        File dir = file.getParentFile().getParentFile();
+        Map<String, String> pipelines= new HashMap<>();
+        dir = new File(dir.getPath().replaceAll("-docker$", "-pipelines"));
```

Comment:
Q: Should the `$` be on the end of `-docker$`?

Reply:
Regex flag to ensure only `-docker` at the end of the path is replaced.